src/main/java/org/generationcp/middleware/dao/dms/GeolocationPropertyDao.java

```diff
@@ -102,18 +102,22 @@ public List<Integer> deleteBlockPropertiesByGeolocationId(final List<Integer> ge
 
 			// block IDs to return to be used for locdes deletion
 			// only return if no other geolocation uses the block
-			SQLQuery sqlQuery1 = this.getSession().createSQLQuery(GET_BLOCK_IDS_TO_DELETE);
+			SQLQuery sqlQuery1 = this.getSession().createSQLQuery(GET_BLOCK_IDS_TO_DELETE)
```

Comment:
This is not very easy to understand, it was not in fact. 
This is only supposed to delete some rows but it is returning some blocks that are not even used internally in the delete script. 
why dont we split this in 2 functions? it is not easy to undertand what the map repesents 
I would find the affected blocks before calling the deletion and if it is out of scope, then much better to return ids and then out of this function do whatever you need to get the affected objects. 
@cheryleigh